Source/WebCore/css/CSSCounterStyle.cpp

```diff
@@ -166,6 +163,183 @@ String CSSCounterStyle::counterForSystemAdditive(unsigned value) const
+        Digit0, Digit1, Digit2, Digit3, Digit4,
+        Digit5, Digit6, Digit7, Digit8, Digit9,
+        NegativeSign
+    };
+
+    if (!number)
+        return { &table[Digit0 - 1] , 1 };
+
+    ASSERT(number != std::numeric_limits<int>::min());
+    bool needsNegativeSign = number < 0;
+    if (needsNegativeSign)
+        number = -number;
+
+    constexpr unsigned groupLength = 8; // 4 digits, 3 digit markers, and a group marker
+    constexpr unsigned bufferLength = 4 * groupLength;
+    AbstractCJKCharacter buffer[bufferLength] = { NoChar };
+
+    for (int i = 0; i < 4; ++i) {
+        int groupValue = number % 10000;
+        number /= 10000;
+
+        // Process least-significant group first, but put it in the buffer last.
+        auto group = &buffer[(3 - i) * groupLength];
+
+        if (groupValue && i)
+            group[7] = static_cast<AbstractCJKCharacter>(SecondGroupMarker - 1 + i);
+
+        // Put in the four digits and digit markers for any non-zero digits.
+        group[6] = static_cast<AbstractCJKCharacter>(Digit0 + (groupValue % 10));
+        if (number || groupValue > 9) {
+            int digitValue = ((groupValue / 10) % 10);
+            group[4] = static_cast<AbstractCJKCharacter>(Digit0 + digitValue);
+            if (digitValue)
+                group[5] = SecondDigitMarker;
+        }
+        if (number || groupValue > 99) {
+            int digitValue = ((groupValue / 100) % 10);
+            group[2] = static_cast<AbstractCJKCharacter>(Digit0 + digitValue);
+            if (digitValue)
+                group[3] = ThirdDigitMarker;
+        }
+        if (number || groupValue > 999) {
+            int digitValue = groupValue / 1000;
+            group[0] = static_cast<AbstractCJKCharacter>(Digit0 + digitValue);
+            if (digitValue)
+                group[1] = FourthDigitMarker;
+        }
+
+        if (formality == Formality::Informal && groupValue < 20) {
+            // Remove the tens digit, but leave the marker.
+            ASSERT(group[4] == NoChar || group[4] == Digit0 || group[4] == Digit1);
+            group[4] = NoChar;
+        }
+
+        if (!number)
+            break;
+    }
+
+    // Convert into characters, omitting consecutive runs of digit0 and trailing digit0.
+    unsigned length = 0;
+    UChar characters[1 + bufferLength];
+    auto last = NoChar;
+    if (needsNegativeSign)
+        characters[length++] = table[NegativeSign - 1];
+    for (unsigned i = 0; i < bufferLength; ++i) {
+        auto character = buffer[i];
+        if (character != NoChar) {
+            if (character != Digit0 || last != Digit0)
+                characters[length++] = table[character - 1];
+            last = character;
+        }
+    }
+    if (last == Digit0)
+        --length;
+
+    return { characters, length };
+}
+
+String CSSCounterStyle::counterForSystemSimplifiedChineseInformal(int value)
+{
+    static constexpr std::array<UChar, 17> simplifiedChineseInformalTable = {
```

Comment:
I think if we leave out the `17` and the `=` the compiler will count for us.

Reply:
I get "too few template arguments" when I remove the 17